pagerduty/user.go

```diff
@@ -299,22 +333,16 @@ func (s *UserService) CreateContactMethod(userID string, contactMethod *ContactM
 
 	resp, err := s.client.newRequestDo("POST", u, nil, &ContactMethodPayload{ContactMethod: contactMethod}, &v)
 	if err != nil {
-		if e, ok := err.(*Error); ok && strings.Compare(fmt.Sprintf("%v", e.Errors), "[User Contact method must be unique]") == 0 {
-			resp, _, lErr := s.ListContactMethods(userID)
-			if lErr != nil {
-				return nil, nil, fmt.Errorf("user contact method is not unique and failed to fetch existing ones: %w", lErr)
-			}
-
-			for _, contact := range resp.ContactMethods {
-				if isSameContactMethod(contact, contactMethod) {
-					return s.GetContactMethod(userID, contact.ID)
-				}
-			}
-
-			return nil, nil, fmt.Errorf("user contact method address is used with different attributes (possibly label)")
-		} else {
+		if e, ok := err.(*Error); !ok || strings.Compare(fmt.Sprintf("%v", e.Errors), "[User Contact method must be unique]") != 0 {
```

Comment:
I asked PD if we can hope to return a 409 here in the future as a more reliable error check, but for now this is probably all we can do.